src/client.cc

```diff
@@ -703,6 +703,8 @@ minio::s3::PutObjectResponse minio::s3::Client::PutObject(PutObjectArgs args) {
     AbortMultipartUpload(amu_args);
   }
 
+  resp.version_id = resp.headers.GetFront("x-amz-version-id");
```

Comment:
1. [CompleteMultipartUpload()](https://github.com/minio/minio-cpp/blob/main/src/baseclient.cc#L338) populates version ID from header properly.
2. Private [PutObject()](https://github.com/minio/minio-cpp/blob/main/src/client.cc#L487) returns `CompleteMultipartUploadResponse` but `PutObjectResponse` takes `Response`. Here you would need to add a constructor to handle it like `CompleteMultipartUploadResponse(const CompleteMultipartUploadResponse& resp) { ... }` or check and typecast base `Response` to `CompleteMultipartUploadResponse`, then handle it properly.